openapi.yaml

```diff
@@ -4346,6 +4350,15 @@ components:
             x-additionalPropertiesName: Dimension Name
             allOf:
               - $ref: '#/components/schemas/dimension'
+        'openeo:property_filters':
+          description: |-
+            Gives a list of properties you can filter on when loading data from the collection.
+            More metadata about the properties can be found in `summaries`.
+
+            Returning an empty array means that no filters are available.
+          type: array
```

Comment:
minor question: should there be a default value specified (empty array)?

Reply:
No, if the default is an empty array then that would mean that by default no data would be returned at all.
The default right now is "not given" => potentially all levels could be included, which was the case before, too.
If "not given" is too implicit we could allow null as a data type and default to that, which means no information available, expect any level to be present. This field anyway is just an indication for clients.

Reply:
thanks, what remains bit unclear from current docs:


- "not given": all properties can be filtered on (backend is expected to raise error when property filter is invalid in some way)?
- side note: is it (useful to make it) possible to make this "everything is allowed, as long as backend doesn't complain" explicit, e.g. by specifying "null"?
- if a list is specified: are only these properties allowed to filter on? Using non-whitelisted properties should raise an error (e.g. client side) ? Or is just guidance for users and are other property filters still allowed?

Reply:
Not giving it doesn't necessarily mean everything is possible. It just is the backward-compatible way of saying: We don't expose it, good luck with figuring it out yourself. ;-)

"Everything is possible" can just be encoded by listing all values explicitly, which I think is the most explicit way. Using null for it is rather confusing from a user POV as this would then put the work onto the user to figure out what is available. And yes, this is meant to be guidance for users (and clients).

Raising errors is part of the processes and would need to be documented there, so we don't imply the best approach for the processes here.